test/test_common/utility.cc

```diff
@@ -23,14 +23,18 @@
 #include "gtest/gtest.h"
 #include "spdlog/spdlog.h"
 
+using testing::GTEST_FLAG(random_seed);
+
 namespace Envoy {
 
-static const uint64_t SEED = std::chrono::duration_cast<std::chrono::nanoseconds>(
-                                 std::chrono::system_clock::now().time_since_epoch())
-                                 .count();
+static const int32_t SEED = std::chrono::duration_cast<std::chrono::nanoseconds>(
+                                std::chrono::system_clock::now().time_since_epoch())
+                                .count();
 
-TestRandomGenerator::TestRandomGenerator() : generator_(SEED) {
-  std::cerr << "TestRandomGenerator running with seed " << SEED;
+TestRandomGenerator::TestRandomGenerator()
+    : generator_(GTEST_FLAG(random_seed) == 0 ? SEED : GTEST_FLAG(random_seed)) {
```

Comment:
We can but it appears to always be 0 as it was when I tried accessing it via flag.
It appears to only be set if you set --gtest_shuffle which I find to be an uncomfortable hidden dependency I'd rather not depend on.

http://docs.ros.org/api/self_test/html/gtest-all_8cc_source.html -> 5752